noun/trace.c

```diff
@@ -300,14 +303,25 @@ u3t_flee(void)
 
 static FILE* trace_file_u = NULL;
 static int nock_pid_i = 0;
+static int trace_counter;
 
 /*  u3t_trace_open(): opens a trace file and writes the preamble.
 */
 void
-u3t_trace_open(c3_c* trace_file_name)
+u3t_trace_open()
 {
-  printf("trace: tracing to %s\n", trace_file_name);
-  trace_file_u = fopen(trace_file_name, "w");
+
+  c3_c fil_c[2048];
+  snprintf(fil_c, 2048, "%s/.urb/put/trace", u3_Host.dir_c);
+
+  struct stat st;
+  if ( -1 == stat(fil_c, &st) ) {
+    mkdir(fil_c, 0700);
+  }
+
+  snprintf(fil_c, 2048, "%s/%ld.json", fil_c, time(NULL));
```

Comment:
I think it'd be better to pretty-print `u3A->wen`, to be consistent with the way |mass effect files are named. Likewise, if we add the ship name there, we should do so here too.

Reply:
nevermind, I found a solution

Reply:
Ok, good. You can't use the loom in any way prior to `u3m_boot()`. Exactly when `u3A->wen` or `u3A->own` is set and available depends on whether it's a first boot or restart. `u3_sist_boot()` is generally the save place to put things that depend on arvo state.

Reply:
we probably want the profiler to catch boot events though right?

Reply:
Yeah, that's probably more important. It'll be much easier to do this right on cc-release.

Reply:
so I can get u3A->wen pretty early in the process, so I'm gonna just name it by that and leave off the ship name. I'll leave off the ship name for |mass as well, for consistency's sake.